lib/fabric/app/models/fabric/refund.rb

```diff
@@ -34,18 +33,19 @@ class Refund
     index({ stripe_id: 1 }, { background: true, unique: true })
 
     def sync_with(refund)
-      self.stripe_id = Fabric.stripe_id_for refund
-      self.charge_id = refund.charge
-      self.payment_intent_id = refund.payment_intent
+      self.stripe_id = stripe_id_for(refund)
+      self.charge_id = handle_expanded(refund.charge)
```

Comment:
You don't need to define fields for `belongs_to` relationships. They were mistakenly left when the relationships were added I guess and throw a warning in the console.

Reply:
The mongoid docs state

> since Mongoid doesn’t know what type of data should be allowed in the field, the field is created with a type of Object. It is a good idea to explicitly define the field with the appropriate type.

I don't think the primary key / foreign key specification is correct here, either. But either way, I think it's preferable to specify the field as String field so it doesn't get stored as a generic Object type.

Reply:
I don't know why the documentation suggests something that will raise warnings. If anything I think it's confusing and non standard to define fields for relationships unless there is an absolute need for it to functionally work.

We'll have a truck load of errors for every field if we do it. They look like this.
`WARN -- : Overwriting existing field latest_invoice_id in class Fabric::Subscription.`

Reply:
Hm, yeah, it's really annoying when those warnings come up. My primary motivation was what the docs say, but that's a pretty strong argument against explicitly defining the fields. The only real implication I see is that if someone did something really stupid and assigned the wrong type of object to that field, there would be no type casting before it gets persisted. Either way though, in that situation, the wrong thing would be persisted; for example, if someone assigned a hash: with a String field type, the hash would be converted to a string, and with an Object field type, it would just be persisted as a hash. I don't think that really makes a significant difference if it's still the wrong data. So yeah, let's forget about that then.